docker/Makefile

```diff
@@ -112,6 +119,23 @@ stop_client :
 # performance requirements are relatively low.
 # -----------------------------------------------------------------
 repository :
+	# If an enclave signing key is available on the host, copy that under build/keys in the repo
+	# Note: the docker build (see PDO_ENCLAVE_CODE_SIGN_PEM in environment.sh) expects the key there
+	if [ ! -z "${PDO_ENCLAVE_CODE_SIGN_PEM}" ]; then \
+		if [ -z "${PDO_SGX_KEY_ROOT}" ]; then \
+			echo "Error: PDO_SGX_KEY_ROOT must be defined to transfer the enclave signing key in docker"; exit 1; \
+		fi ;\
+		if [ -e ${PDO_ENCLAVE_CODE_SIGN_PEM} ]; then \
+			echo "Enclave code signing key: using ${PDO_ENCLAVE_CODE_SIGN_PEM}"; \
+			(test ${PDO_ENCLAVE_CODE_SIGN_PEM} -ef ${PDO_SGX_KEY_ROOT}/enclave_code_sign.pem || \
+			cp ${PDO_ENCLAVE_CODE_SIGN_PEM} ${PDO_SGX_KEY_ROOT}/enclave_code_sign.pem); \
+		else \
+			echo "Enclave code signing key: unavailable, a new one will be generated"; \
+		fi \
+	else \
+		echo "Enclave code signing key: PDO_ENCLAVE_CODE_SIGN_PEM undefined, a new one will be generated"; \
+	fi
+	# clone the repo
 	git clone --single-branch --branch $(PDO_BRANCH) --recurse-submodules '$(PDO_REPO)' repository
```

Comment:
this is way too much logic to put in a makefile. especially since it could be useful for figuring out the keys even beyond docker. please move this into a separate script (build/__tools__/copy_sgx_keys.sh?).

Reply:
That's reasonable. Since this is a docker tool, I'll put it under `docker/tools/copy_sgx_collateral.sh`.